### PR TITLE
 [Communication - Phone Number Service] Proposed countries offerings API for the ACS Phone Number Service.

### LINKED ISSUE #18871: [Communication - Phone Number Service] REST API Review
Service name:
ACS - Phone Number Service

Key contact for the review:
adrian.tang@microsoft.com

Whether this is a new or existing API:
Existing API

PR
https://github.com/Azure/azure-rest-api-specs/pull/18869

Previous REST.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/PhoneNumbers/stable/2021-03-07/phonenumbers.json

[Hero scenarios](https://skype.visualstudio.com/SPOOL/_wiki/wikis/SPOOL.wiki/31514/Hero-Scenarios)

There are four new proposed APIs:
API that returns the list of eligible countries for the Azure resource.
API that returns the list of offerings given a country.
API that returns the list of available cities with phone numbers available, given a country.
API that returns the list of available area codes, given a city.

### PR DESCRIPTION
**Pull request by whisper6284**
This is a draft of the proposed countries offerings API for the ACS Phone Number Service.

The following is the list of new APIs proposed for the ACS Phone Number Service.

- Countries. Returns the list of supported and eligible countries.
- Localities. Returns the list of available cities / towns (or as we will call them, localities) of phone numbers.
- Area Codes. Returns the list of available area codes. In the case of Toll-Free, the area codes are for the country. For Geographic area codes, a locality (city, or town) must be provided.
- Offerings. Returns the list of offers (with rates) of the capabilities combinations offered.

### CODE REVIEW

**Comment by mikekistler** [comment]
You need to update the readme.md file with a new tagged version to reference the version in this PR and make this the default tag.  That will address the Avocado errors.

File: specification/communication/data-plane/PhoneNumbers/readme.md
Comment (mikekistler) [inline specification/communication/data-plane/PhoneNumbers/readme.md:29]: I think you also need this change:
```suggestion
tag: package-phonenumber-2022-03-01
```

**Comment by markweitzel** [comment]
@whisper6284 -- There are some "cleanliness" errors that should be fixed up, e.g. 
* `Error response should contain a x-ms-error-code header.`
* `Operation does not define an "api-version" query parameter.`
* `Operations with a 202 response should specify x-ms-long-running-operation: true.`
* `OperationId for delete method should contain "Delete"`
* `Response body schema of pageable response should contain top-level array property value`

You can easily identify these by running the Spectral linter, which is available as part of our [REST API Style Guide](https://aka.ms/style).

In addition, you should review the generated developer docs that are produced as part of the pipeline.

File: specification/communication/data-plane/PhoneNumbers/stable/2022-03-31/phonenumbers.json
Comment (mikekistler) [inline specification/communication/data-plane/PhoneNumbers/stable/2022-03-31/phonenumbers.json:122]: The trailing comma here is tripping up the pipeline checks.
```suggestion
          }
```

**Review by markweitzel — APPROVED** [review]
Per #18869 -- This is approved pending minor cleanup on the swagger.

**Review by lmazuel — APPROVED** [review]
Suppression is necessary based on already released SDK.

**Comment by konrad-jamrozik** [comment]
@lmazuel @JeffreyRichter supposedly the suppression is not working as expected, I filed an issue to track this:
- https://github.com/Azure/azure-sdk-tools/issues/6044

Can you override-merge this PR in spite of that? Thanks!

**Pull request merged**
